### PR TITLE
[Doc] [KubeRay] Update `v0.6.0` to `v1.0.0-rc.0`

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Analogous to https://github.com/ray-project/ray/pull/37833, this PR updates the default KubeRay version in the docs to 1.0.0-rc.0.  We will release KubeRay `1.0.0-rc.0` before the Ray 2.7 docs go live.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by kevin85421 — COMMENTED** [review]
Could you check all `0.6.0` appearances under `doc/source/cluster/kubernetes`? For example, `ingress.md` installs KubeRay operator v0.6.0 and RayCluster v0.6.0, but the doc is not changed by this PR. Thanks!

**Comment by architkulkarni** [comment]
@kevin85421 ah thanks I forgot to save that file, could you review it again?

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md [hunk 1]
```diff
@@ -73,70 +72,6 @@ kubectl describe ingress ray-cluster-ingress
 kubectl delete ingress ray-cluster-ingress
 ```
 
-(kuberay-gke-ingress)=
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:76]: why do we remove this?

**Comment by architkulkarni** [comment]
Oh sorry I must have messed up my merge let me take a look

**Comment by architkulkarni** [comment]
@kevin85421 Fixed, sorry for the unnecessary back and forth

**Review by kevin85421 — APPROVED** [review]

**Comment by architkulkarni** [comment]
Windows failure `serve:test_autoscaling_policy ` unrelated

**Pull request merged**